src/posit/connect/oauth/sessions.py

```diff
@@ -2,34 +2,101 @@
 
-class Session(Resource):
+
+class Session(ActiveDict):
     """OAuth session resource."""
 
+    class _Attrs(TypedDict, total=False):
+        id: str
+        "The internal numeric identifier of this OAuth session."
+        guid: str
+        "The unique identifier of this OAuth session which is used in REST API requests."
+        user_guid: str
+        "The unique identifier of the user that is associated with this OAuth session."
+        oauth_integration_guid: str
+        "The unique identifier of the OAuth integration that is associated with this OAuth session."
+        has_refresh_token: bool
+        "Indicates whether this OAuth session has a refresh token."
+        created_time: str
+        "The timestamp (RFC3339) indicating when this OAuth session was created."
+        updated_time: str
+        "The timestamp (RFC3339) indicating when this OAuth session was last updated."
+
+    @overload
+    def __init__(self, ctx: Context, /, *, guid: str) -> None: ...
+
+    """
+    Retrieve an OAuth session by its unique identifier.
+
+    Parameters
+    ----------
+    ctx : Context
+        The context object containing the session and URL for API interactions.
+    guid : str
+        The unique identifier of the OAuth session.
+    """
+
+    @overload
+    def __init__(self, ctx: Context, /, **kwargs: Unpack["Session._Attrs"]) -> None: ...
+
+    """
+    Retrieve an OAuth session by its unique identifier.
+
+    Parameters
+    ----------
+    ctx : Context
+        The context object containing the session and URL for API interactions.
+    **kwargs : Session._Attrs
+        Attributes for the OAuth session. If not supplied, the attributes will be retrieved from the API upon initialization.
+    """
+
+    def __init__(self, ctx: Context, /, **kwargs) -> None:
+        guid = assert_guid(kwargs.get("guid"))
+        path = self._api_path(guid)
+
+        # Only fetch data if `kwargs` only contains `"guid"`
+        get_data = len(kwargs) == 1
+
+        super().__init__(ctx, path, get_data, **kwargs)
+
+    # TODO-barret-future-q: Should this be destroy?
     def delete(self) -> None:
-        path = f"v1/oauth/sessions/{self['guid']}"
-        url = self.params.url + path
-        self.params.session.delete(url)
+        """Destroy the OAuth session."""
+        self._delete_api()
+
+    @classmethod
+    def _api_path(cls, session_guid: str) -> str:
+        return f"v1/oauth/sessions/{session_guid}"
+
 
+class Sessions:
+    def __init__(self, ctx: Context) -> None:
+        super().__init__()
+        self._ctx = ctx
 
-class Sessions(Resources):
+    # TODO-barret-future-q: Should this be `.all()`?
     @overload
     def find(
         self,
         *,
         all: Optional[bool] = ...,
     ) -> List[Session]: ...
 
+    # TODO-barret-future-q: Should this be `.find_by()`?
     @overload
     def find(self, **kwargs) -> List[Session]: ...
 
     def find(self, **kwargs) -> List[Session]:
```

Comment:
No. `all: bool` is a query parameter. Instead, it should be `**kwargs: Unpack[Sessions._FindAttrs]`